HeterogeneousCore/MPIServices/plugins/MPIService.cc

```diff
@@ -0,0 +1,74 @@
+// -*- C++ -*-
+
+#include <mpi.h>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/ActivityRegistry.h"
+#include "FWCore/Utilities/interface/Exception.h"
+
+namespace {
+  const char* mpi_thread_support_level[] = {
+      "MPI_THREAD_SINGLE",      // only one thread will execute (the process is single-threaded)
+      "MPI_THREAD_FUNNELED",    // only the thread that called MPI_Init_thread will make MPI calls
+      "MPI_THREAD_SERIALIZED",  // only one thread will make MPI library calls at one time
+      "MPI_THREAD_MULTIPLE"     // multiple threads may call MPI at once with no restrictions
+  };
+}
+
+class MPIService {
+public:
+  MPIService(edm::ParameterSet const& config, edm::ActivityRegistry& registry);
```

Comment:
`ActivityRegistry` parameter looks unused, so I'd remove it.

Reply:
It doesn't seem possible to have the constructor without the `ActivityRegistry` parameter:
```
>> Compiling edm plugin /data/user/fwyzard/patatrack/CMSSW_11_2_0_pre3/src/HeterogeneousCore/MPIServices/plugins/MPIService.cc
In file included from /data/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-bcolbf/include/c++/8.4.0/memory:80,
                 from /data/cmssw/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_0_pre3/src/FWCore/MessageLogger/interface/MessageLogger.h:128,
                 from /data/user/fwyzard/patatrack/CMSSW_11_2_0_pre3/src/HeterogeneousCore/MPIServices/plugins/MPIService.cc:5:
/data/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-bcolbf/include/c++/8.4.0/bits/unique_ptr.h: In instantiation of 'typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = MPIService; _Args = {const edm::ParameterSet&, edm::ActivityRegistry&}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<MPIService, std::default_delete<MPIService> >]':
/data/cmssw/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_0_pre3/src/FWCore/ServiceRegistry/interface/ServiceMaker.h:49:43:   required from 'std::unique_ptr<_Tp> edm::serviceregistry::AllArgsMaker<T, TConcrete>::make(const edm::ParameterSet&, edm::ActivityRegistry&) const [with T = MPIService; TConcrete = MPIService]'
/data/cmssw/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_0_pre3/src/FWCore/ServiceRegistry/interface/ServiceMaker.h:78:28:   required from 'bool edm::serviceregistry::ServiceMaker<T, TMaker>::make(const edm::ParameterSet&, edm::ActivityRegistry&, edm::serviceregistry::ServicesManager&) const [with T = MPIService; TMaker = edm::serviceregistry::AllArgsMaker<MPIService, MPIService>]'
/data/cmssw/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_0_pre3/src/FWCore/ServiceRegistry/interface/ServiceMaker.h:76:12:   required from here
/data/cmssw/slc7_amd64_gcc820/external/gcc/8.2.0-bcolbf/include/c++/8.4.0/bits/unique_ptr.h:835:30: error: no matching function for call to 'MPIService::MPIService(const edm::ParameterSet&, edm::ActivityRegistry&)'
     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/data/user/fwyzard/patatrack/CMSSW_11_2_0_pre3/src/HeterogeneousCore/MPIServices/plugins/MPIService.cc:31:1: note: candidate: 'MPIService::MPIService(const edm::ParameterSet&)'
 MPIService::MPIService(edm::ParameterSet const& config) {
 ^~~~~~~~~~
/data/user/fwyzard/patatrack/CMSSW_11_2_0_pre3/src/HeterogeneousCore/MPIServices/plugins/MPIService.cc:31:1: note:   candidate expects 1 argument, 2 provided
/data/user/fwyzard/patatrack/CMSSW_11_2_0_pre3/src/HeterogeneousCore/MPIServices/plugins/MPIService.cc:21:7: note: candidate: 'constexpr MPIService::MPIService(const MPIService&)'
 class MPIService {
       ^~~~~~~~~~
/data/user/fwyzard/patatrack/CMSSW_11_2_0_pre3/src/HeterogeneousCore/MPIServices/plugins/MPIService.cc:21:7: note:   candidate expects 1 argument, 2 provided
gmake: *** [config/SCRAM/GMake/Makefile.rules:1678: tmp/slc7_amd64_gcc820/src/HeterogeneousCore/MPIServices/plugins/HeterogeneousCoreMPIServicesAuto/MPIService.cc.o] Error 1
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```